code/src/components/TodoList.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+
+import { useSelector, useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { todos } from '../reducers/todos';
+import EmptyTodoList from './EmptyTodoList';
+import TodoItem from './TodoItem';
+import ActionButtons from './ActionButtons';
+import { FilterSelect } from '../lib/Select';
+
+const TodoList = () => {
+  const dispatch = useDispatch();
+  const list = useSelector(store => store.todos.list.items);
+  const filteredTodoStatus = useSelector(store => store.todos.list.selectedFilterStatus);
+  const filteredTodoCategory = useSelector(store => store.todos.list.selectedFilterCategory);
+  const categories = useSelector((store) => store.todos.list.categories);
+
+  const filteredTodosStatus = (value) => {
+    dispatch (
+      todos.actions.filteredStatus(value)
+    );
+  };
+
+  const filteredTodosCategory = (value) => {
+    dispatch (
+      todos.actions.filteredCategory(value)
+    );
+  };
+
+  const filteredList = list.filter(item => {
+    if (filteredTodoCategory) {
+      if (filteredTodoStatus === 'not complete') {
+        return item.category === filteredTodoCategory && !item.complete;
+      } else if (filteredTodoStatus === 'complete') {
+        return item.category === filteredTodoCategory && item.complete;
+      } else return item.category === filteredTodoCategory;
+    } else if (filteredTodoStatus === 'complete') {
+      return item.complete;
+    } else if (filteredTodoStatus === 'not complete') {
+      return !item.complete;
+    } else return item;
+  });
```

Comment:
This looks like it was hard work! Impressive

Reply:
It was a struggle 😅